### PR TITLE
Bdanek/OMOP + eICU test + stat test + Event class default values + Github actions

### PR DESCRIPTION
**Pull request by BPDanek**
1. Pyhealth default fields on Event
2. eICU test against local data (no API yet)
3. EHR base dataset stat test (prevent redundant repeating per class)

Edit: merged https://github.com/sunlabuiuc/PyHealth/pull/114 in, including adding 
4. OMOP test
5. github action triggered manually: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
(go to "actions" (besides discussions and pull requests) and trigger the python unit test workflow)
example of actions: https://github.com/BPDanek/PyHealth/actions/runs/4259851645/jobs/7412447994

### CODE REVIEW

File: pyhealth/data/data.py [hunk 1]
```diff
@@ -41,11 +41,11 @@ class Event:
...
-        table: str,
-        vocabulary: str,
-        visit_id: str,
-        patient_id: str,
+        code: str = None,
+        table: str = None,
+        vocabulary: str = None,
+        visit_id: str = None,
+        patient_id: str = None,
         timestamp: Optional[datetime] = None,
```
Comment (BPDanek) [inline pyhealth/data/data.py:49]: @ycq091044 is this going to cause issues? Running the Event unit test suite indicates no, but I am not sure.

Comment (ycq091044) [reply pyhealth/data/data.py:49]: I think it is fine.

File: pyhealth/unittests/test_datasets/test_eicu.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+from pyhealth.data.data import Event
+
+from pyhealth.datasets import eICUDataset
+from pyhealth.unittests.test_datasets.utils import EHRDatasetStatAssertion
+
+
+class eICUDataset(unittest.TestCase):
+
+    # to test the file this path needs to be updated
+    ROOT = "~/Code/PyHealth_sunlab/PyHealth/datasets/eicu-collaborative-research-database-demo-2.0.1"
```
Comment (ycq091044) [inline pyhealth/unittests/test_datasets/test_eicu.py]: emmm, I will let patrick to help upload this demo version https://physionet.org/content/eicu-crd-demo/2.0.1/.

**Review by ycq091044 — COMMENTED** [review]
The code looks good after a glance. Will test it locally.

File: pyhealth/unittests/test_datasets/test_eicu.py (hunk 2 above)
Comment (ycq091044) [reply pyhealth/unittests/test_datasets/test_eicu.py]: Thanks @pat-jj. Hi benjamin, please try this https://storage.googleapis.com/pyhealth/eicu-demo/

Comment (BPDanek) [reply pyhealth/unittests/test_datasets/test_eicu.py]: I forgot to post the source: https://physionet.org/content/eicu-crd-demo/2.0.1/

**Comment by ycq091044** [comment]
Hi @BPDanek , we have uploaded the data "https://physionet.org/content/eicu-crd-demo/2.0.1/" to googleapis. I updated the link in the PR. Could you please correct the eicu unittest information. It does not seem to run through.

**Comment by BPDanek** [comment]
@ycq091044 @pat-jj updated

**Comment by ycq091044** [comment]
Hi @BPDanek , I checked your PR locally and added the "refresh_cache=True" and the "dataset_name" for all dataset unittest.

I did not use "refresh_cache=True" initially, and the unittest_eicu failed. Maybe my cached version is not the same version. So, let us use "refresh_cache=True" to always update users' cache (it is ok for the demo dataset).

I use xxx-demo as the dataset_name. If "refresh_cache=True" is turned on, the demo dataset (by default, using the function name as the dataset name) may overwrite the non-demo datasets.

**Comment by BPDanek** [comment]
Test results: https://github.com/sunlabuiuc/PyHealth/actions/runs/4276364884

**Pull request merged**